src/plugins/user-agent-protection.js

```diff
@@ -0,0 +1,25 @@
+/**
+ * User-Agent protection plugin to mitigate ReDoS attacks
+ * Protects against GHSA-mgfv-m47x-4wqp vulnerability in useragent package
+ * Truncates User-Agent headers to prevent malicious payloads
+ */
```

Comment:
Attack is that a long string is added to `useragent` header that causes overly complicated parsing risking CPU overload.

As `useragent` seems to be no longer supported, then automatically truncate unreasonably long `user-agent` headers.